### PR TITLE
Fixed unit tests

### PR DESCRIPTION
**Pull request by drolando**
Fixed unit tests:
- [error method](https://github.com/Yelp/statsd/compare/master...Yelp:fix_tests?expand=1#diff-da4d9dfd9cd6f8f204f7b0cf10f411f3R37) missing in the backendLogger object
- [`this.__proto__`](https://github.com/Yelp/statsd/compare/master...Yelp:fix_tests?expand=1#diff-57565560c47cf7e6f85ea7442c65e724L25) prototype modification after object initialization
- reverted run_test.sh to the default one
- [`log.__proto__`](https://github.com/Yelp/statsd/compare/master...Yelp:fix_tests?expand=1#diff-da4d9dfd9cd6f8f204f7b0cf10f411f3L107) was changing the global logger

Reverted `graphite.js` backend to the default one.

Moved the backendLogger creation inside the `multidimensional_graphite.js` backend to avoid changes to the default `stats.js`.

Changed a couple of variable and function names to follow statsd style guide lines.

Restored etsy/statsd files permissions.

### CODE REVIEW

**Comment by poros** [comment]
Diff between master in etsy/statsd and @drolando 's branch
https://github.com/etsy/statsd/compare/master...Yelp:fix_tests

File: backends/multidimensional_graphite.js [hunk 1]
```diff
@@ -1,10 +1,47 @@
...
+  , events = require('events')
+  , logger = require('../lib/logger')
+  , getHostname = require('os').hostname
+  , getBasename = require('path').basename;
 
 var log = console; // the logger
 var debug;
 
 
+function  createBackendLogger(config, logger) {
```
Comment (poros) [inline backends/multidimensional_graphite.js:11]: I strongly support this change. Modifying the global logger was causing a lot of tests to fail and it's something you shouldn't really do after the initialization carried in the main entry point of the application. Moving this change to our custom backend file also isolates our code changes from the main codebase and is going to help us avoiding merge conflicts when updating or when pushing patches upstream (if ever).

**Comment by poros** [comment]
Overall diff with etsy/statsd master is much more sane than before.
The refactored code seems legit to me, but I don't know much about the original logic. I assume it is working as intended if unit tests pass.
+1 but wait for @zachm and @squirrely before merging, so they can voucher your changes didn't impact the logic.

File: backends/multidimensional_graphite.js (hunk 1 above)
Comment (rybit) [inline backends/multidimensional_graphite.js:11]: I think that we should push this into the lib/logging and do it by default. At least the debug/info/warn parts.

Comment (rybit) [inline backends/multidimensional_graphite.js:11]: also isn't this taken care of in the modification that is in the stats.js? It creates a logger with this ability for each of the backends

**Comment by drolando** [comment]
@squirrely I reverted it to avoid as much as possible modifications in the etsy/statsd code.
If you want to run only one test you can use `nodeunit <test_file>`, so that change to the script is not a  really useful improvement.

**Comment by rybit** [comment]
after talking with @drolando he informed me I can just do a `nodeunit <test>` or `nodeunit <test> -f <spec>`. This removes my concern with that!

**Comment by drolando** [comment]
I'm going to merge this branch into master to fix the tests and get a green build on travis-ci.
If you have any more concerns about the changes that I made we can fix in a later commit

**Pull request merged**